python-stdlib/logging/logging.py

```diff
@@ -86,43 +135,92 @@ def error(self, msg, *args):
     def critical(self, msg, *args):
         self.log(CRITICAL, msg, *args)
 
-    def exc(self, e, msg, *args):
+    def exception(self, msg, *args):
         self.log(ERROR, msg, *args)
-        sys.print_exception(e, _stream)
+        if hasattr(sys, "exc_info"):
+            for h in filter(lambda h: isinstance(h, StreamHandler), self.handlers):
+                sys.print_exception(sys.exc_info()[1], h.stream)
 
-    def exception(self, msg, *args):
-        self.exc(sys.exc_info()[1], msg, *args)
+    def addHandler(self, handler):
+        self.handlers.append(handler)
 
-    def addHandler(self, hndlr):
-        self.handlers.append(hndlr)
+    def hasHandlers(self):
+        return len(self.handlers) > 0
 
 
-_level = INFO
-_loggers = {}
+def info(msg, *args):
+    getLogger().log(INFO, msg, *args)
 
 
-def getLogger(name="root"):
-    if name in _loggers:
-        return _loggers[name]
-    l = Logger(name)
-    _loggers[name] = l
-    return l
+def debug(msg, *args):
+    getLogger().log(DEBUG, msg, *args)
 
 
-def info(msg, *args):
-    getLogger().info(msg, *args)
+def warning(msg, *args):
+    getLogger().log(WARNING, msg, *args)
 
 
-def debug(msg, *args):
-    getLogger().debug(msg, *args)
-
-
-def basicConfig(level=INFO, filename=None, stream=None, format=None):
-    global _level, _stream
-    _level = level
-    if stream:
-        _stream = stream
-    if filename is not None:
-        print("logging.basicConfig: filename arg is not supported")
-    if format is not None:
-        print("logging.basicConfig: format arg is not supported")
+def error(msg, *args):
+    getLogger().log(ERROR, msg, *args)
+
+
+def critical(msg, *args):
+    getLogger().log(CRITICAL, msg, *args)
```

Comment:
For these functions (warning, error, info etc), it would generate slightly smaller code with how it was originally, ie: `getLogger().critical(msg, *args)` etc.